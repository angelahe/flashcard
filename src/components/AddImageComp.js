import React from 'react';
import PropTypes from 'prop-types';
import searchbtn from '../img/buttons/search_FFFFFF.png';
import searchresults from './testsearchresults';
import Thumbnail from './Thumbnail';

class AddImageComp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      term: '',
      currentImageUrl: 'https://d30y9cdsu7xlg0.cloudfront.net/png/75112-200.png',
      currentImageId: 75112,
      imagesList: [],
    };
  }

  handleTermChange = (event) => {
    this.setState({ term: event.target.value });
  }

  handleSearchClick = () => {
    const { term } = this.state;

    const results = searchresults.searchresults;

    console.log('results is', results);
    this.setState({ imagesList: results });

    // call the nounproject api with the search term
    console.log(`going to search for ${term} on nounproject`);
  }

  handleImageSend = () => {
    const { currentImageUrl, currentImageId } = this.state;
    const { onImageAdded } = this.props;
    console.log('in handleImageSend');
    onImageAdded(currentImageUrl, currentImageId);
  }

  // temp able to enter the id and url by user
  handleImageUrlChange = (event) => {
    this.setState({ currentImageUrl: event.target.value });
  }

  handleImageIdChange = (event) => {
    this.setState({ currentImageId: event.target.value });
  }

  handleImageSelect = (imageUrl, imageId) => {
    console.log(`passed in image ${imageUrl} and ${imageId}`);
    this.setState({ currentImageUrl: imageUrl, currentImageId: imageId });
  }

  render() {
    const {
      term,
      currentImageId,
      currentImageUrl,
      imagesList,
    } = this.state;

    const imagesListItems = imagesList.map(image => (
      <Thumbnail
        key={image.id}
        imageUrl={image.href}
        imageId={image.id}
        onImageSelect={this.handleImageSelect}
      />
    ));
    return (
      <div>
        <span>Image Search:</span>
        <input value={term} onChange={this.handleTermChange} />
        <button type="button" onClick={this.handleSearchClick}>
          <img className="btnImg" src={searchbtn} alt="Add" />
        </button>
        <br />
        <div className="cardContainer">
          <img className="cardImg" src={currentImageUrl} alt="nounprojectimg" />
        </div>
        <div className="imagesContainer">
          {imagesListItems}
        </div>
        <p>Image Selected Values</p>
        <span>Url: {currentImageUrl} ID: {currentImageId}</span>
        <button type="button" onClick={this.handleImageSend}>
          OK
        </button>
      </div>
    );
  }
}

AddImageComp.defaultProps = {
  onImageAdded: () => { },
};

AddImageComp.propTypes = {
  onImageAdded: PropTypes.func,
};

export default AddImageComp;
