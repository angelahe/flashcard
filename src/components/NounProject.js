function getImages(term) {
  return fetch(`icon?term=${term}`, {
    method: 'GET',
  })
    .then(response => response.json());
}
const NounProject = { getImages };
export default NounProject;
