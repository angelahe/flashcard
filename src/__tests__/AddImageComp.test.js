import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddImageComp from '../components/AddImageComp';
import NounProject from '../components/NounProject';
import searchresults from '../testdata/testsearchresults';

it('renders without crashing', () => {
  shallow(<AddImageComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddImageComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('AddCardComp renders correctly', () => {
  const tree = renderer.create(<AddImageComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div AddImageComp on shallow render', () => {
  const component = shallow(<AddImageComp />);
  expect(component.html()).toContain('AddImageComp');
  expect(component.html()).toContain('SearchTerm');
  expect(component.html()).toContain('SearchBtn');
  expect(component.html()).toContain('ImagePreview');
  expect(component.html()).toContain('DoneBtn');
  expect(component.html()).toContain('Thumbnails');
});

it('should simulate a change to SearchTerm input field', (done) => {
  const component = shallow(<AddImageComp />);
  const searchTerm = component.find('.SearchTerm').first();
  searchTerm.simulate('change', { target: { value: 'cat' } });
  setTimeout(() => {
    expect(component.instance().state.term).toBe('cat');
    done();
  });
});

it('should simulate a click on search button', (done) => {
  const handleImageAdded = jest.fn(() => {});
  const results = searchresults.searchresults;
  NounProject.getImages = jest.fn(() => Promise.resolve(results));

  const component = shallow(<AddImageComp onImageAdded={handleImageAdded} />);
  const searchTerm = component.find('.SearchTerm').first();
  searchTerm.simulate('change', { target: { value: 'cat' } });
  const button = component.find('.SearchBtn').first();
  button.simulate('click');
  setTimeout(() => {
    expect(component.instance().state.imagesList).toBe(results);
    done();
  });
});

it('should simulate a click on Done button', (done) => {
  const currentImageUrl = 'https://d30y9cdsu7xlg0.cloudfront.net/png/77680-200.png';
  const currentImageId = '77680';
  const handleImageAdded = jest.fn(() => {});
  const component = shallow(<AddImageComp onImageAdded={handleImageAdded} />);
  const button = component.find('.DoneBtn').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleImageAdded.mock.calls.length).toBe(1);
    expect(handleImageAdded.mock.calls[0][0]).toBe(currentImageUrl);
    expect(handleImageAdded.mock.calls[0][1]).toBe(currentImageId);
    done();
  });
});
