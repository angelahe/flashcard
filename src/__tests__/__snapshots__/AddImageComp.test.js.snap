// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCardComp renders correctly 1`] = `
<div
  className="AddImageComp"
>
  <div
    className="LineContainer"
  >
    <input
      className="SearchTerm DetailText"
      onChange={[Function]}
      placeholder="Find an image"
      value=""
    />
    <button
      className="SearchBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add"
        className="btnImg"
        src="search_FFFFFF.png"
      />
    </button>
  </div>
  <br />
  <div
    className="LineContainer"
  >
    <div
      className="ImagePreview"
    >
      <img
        alt="nounprojectimg"
        className="cardImg"
        src="https://d30y9cdsu7xlg0.cloudfront.net/png/77680-200.png"
      />
    </div>
    <button
      className="DoneBtn AppBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Done"
        className="btnImg"
        src="done_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="Thumbnails imagesContainer"
  />
  <p>
    Image Selected Values
  </p>
  <span>
    Url: 
    https://d30y9cdsu7xlg0.cloudfront.net/png/77680-200.png
     ID: 
    77680
  </span>
</div>
`;
